internal/xds/translator/listener.go

```diff
@@ -71,6 +73,61 @@ func buildXdsListener(httpListener *ir.HTTPListener) (*listener.Listener, error)
 	}, nil
 }
 
+func buildXdsPassthroughListener(clusterName string, tcpListener *ir.TCPListener) (*listener.Listener, error) {
+	if tcpListener == nil {
+		return nil, errors.New("http listener is nil")
+	}
+
+	mgr := &tcp.TcpProxy{
+		StatPrefix: "passthrough",
+		ClusterSpecifier: &tcp.TcpProxy_Cluster{
+			Cluster: clusterName,
+		},
+	}
+	mgrAny, err := anypb.New(mgr)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
not a blocker but should `tcpListener.SNIs` be replaced with `tcpListener.TLS.SNIs`, so the code under this can include the TLS inspector if `tcpListener.TLS != nil`